app/models/post.rb

```diff
@@ -5,5 +5,16 @@ class Post < ApplicationRecord
 
   validates :title, presence: true
   validates :text, presence: true
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- **[OPTIONAL]** We can add a callback trigger for this method so that it is called whenever a post is created. So we won't have to call it ourselves everytime we create a post. You can add the `after_save` active record callback at the top of the class.

  ```
  after_save :update_user_posts_counter

  .
  .
  .

  private

  def update_user_posts_counter
    author.update(posts_counter: author.posts.count)
  end
  ```